autoload/codefmt/clangformat.vim

```diff
@@ -55,6 +55,36 @@ function! s:ClangFormatHasAtLeastVersion(minimum_version) abort
 endfunction
 
 
+" Inputs are 1-based (row, col) coordinates into lines.
+" Returns the corresponding zero-based offset into lines->join("\n")
+function! s:PositionToOffset(row, col, lines) abort
+  let l:offset = a:col - 1
+  if a:row > 1
+    for l:line in a:lines[0 : a:row - 2]
+      let l:offset += len(l:line) + 1
+    endfor
+  endif
+  return l:offset
+endfunction
+
+
+" Input is zero-based offset into lines->join("\n")
+" Returns the 1-based [row, col] coordinates into lines.
+function! s:OffsetToPosition(offset, lines) abort
+  let l:lines_consumed = 0
+  let l:chars_left = a:offset
+  for l:line in a:lines
+    let l:chars_after_next = l:chars_left - len(l:line) - 1
+    if l:chars_after_next < 0
+      break
+    endif
+    let l:chars_left = l:chars_after_next
```

Comment:
Up to you, but here's an equivalent that's slightly easier to read for me, flipping the `l:chars_after_next` math:
```suggestion
    let l:line_len = len(l:line) + 1
    if l:chars_left < l:line_len
      break
    endif
    let l:chars_left -= l:line_len
```